02week/ticTacToe.js

```diff
@@ -29,59 +29,115 @@ function printBoard() {
 
+// should return true, if the player won on any row
 function horizontalWin() {
-  for (let i = 0; i <= 2; i++) {
-    if (
-      board[i][0] == playerTurn &&
-      board[i][1] == playerTurn &&
-      board[i][2] == playerTurn
-    ) {
-      return true;
-    }
-    return false;
+  let hWin = board[0][0];
+  if (
+    (board[0][0] == playerTurn &&
+      board[0][1] == playerTurn &&
+      board[0][2] == playerTurn) ||
+    (board[1][0] == playerTurn &&
+      board[1][1] == playerTurn &&
+      board[1][2] == playerTurn) ||
+    (board[2][0] == playerTurn &&
+      board[2][1] == playerTurn &&
+      board[2][2] == playerTurn)
+  ) {
+    return true;
   }
 }
 
+horizontalWin();
+
+// function horizontalWin() {
+//   for (let i = 0; i <= 2; i++) {
+//     if (
+//       board[i][0] == playerTurn &&
+//       board[i][1] == playerTurn &&
+//       board[i][2] == playerTurn
+//     ) {
+//       return true;
+//     }
+//     // else {
+//     //   return false;
+//     // }
+//   }
+// }
+
+// should return true, if the player won on any column
 function verticalWin() {
-  for (let i = 0; i <= 2; i++) {
-    if (
-      board[0][i] == playerTurn &&
-      board[1][i] == playerTurn &&
-      board[2][i] == playerTurn
-    ) {
-      return true;
-    }
-    return false;
+  let vWin = board[0][0];
+  if (
+    (board[0][0] == playerTurn &&
+      board[1][0] == playerTurn &&
+      board[2][0] == playerTurn) ||
+    (board[0][1] == playerTurn &&
+      board[1][1] == playerTurn &&
+      board[2][1] == playerTurn) ||
+    (board[0][2] == playerTurn &&
+      board[1][2] == playerTurn &&
+      board[2][2] == playerTurn)
+  ) {
+    return true;
   }
 }
 
+verticalWin();
+
+// should return true, if the player won on any diagonal
 function diagonalWin() {
+  let dWin = [0][0];
   if (
     (board[0][0] == playerTurn &&
       board[1][1] == playerTurn &&
       board[2][2] == playerTurn) ||
-    (board[2][0] == playerTurn &&
+    (board[0][0] == playerTurn &&
       board[1][1] == playerTurn &&
-      board[0][2] == playerTurn)
+      board[2][2] == playerTurn)
   ) {
     return true;
   }
-  return false;
 }
 
+diagonalWin();
+
+// should return true if the player won
+// (if any of the top 3 functions return true, this method should return true)
 function checkForWin() {
-  if (horizontalWin || verticalWin || diagonalWin) {
-    console.log("Player X Wins!");
+  if (horizontalWin()) {
+    return true;
+  } else if (verticalWin()) {
+    return true;
+  } else if (diagonalWin()) {
+    return true;
   }
 }
+// checkForWin(playerTurn);
+// console.log(playerTurn);
+//   } else {
+// console.log(checkForWin(playerTurn, "Congrats!"));
 
+// console.log(checkForWin);
+//   }
+// }
+
+// set the value on that box
+// check if the player won
+// if they won (say congrats!)
+// switch to player 2
 function ticTacToe(row, column) {
-  board[row][column] = playerTurn;
+  if (playerTurn == "X") {
```

Comment:
would be nice to check if the move is valid.
Also, dont see where the checkForWin is called.